src/Features/Core/Portable/InlineMethod/AbstractInlineMethodRefactoringProvider.MethodParametersInfo.cs

```diff
@@ -0,0 +1,297 @@
+                    .ToImmutableArray();
+
+                // 4. All the remaining arguments, which might includes method call and a lot of other expressions.
+                // Generate a declaration in the caller.
+                // Example:
+                // Before:
+                // void Caller(bool x)
+                // {
+                //     Callee(Foo(), x ? Foo() : Bar())
+                // }
+                // void Callee(int a, int b)
+                // {
+                //     DoSomething(a, b);
+                // }
+                // After:
+                // void Caller(bool x)
+                // {
+                //     int a = Foo();
+                //     int b = x ? Foo() : Bar();
+                //     DoSomething(a, b);
+                // }
+                // void Callee(int a, int b)
+                // {
+                //     DoSomething(a, b);
+                // }
+                var parametersNeedGenerateDeclarations = parameterSymbolAndArguments
+                    .RemoveRange(parametersWithIdentifier)
+                    .RemoveRange(parametersWithVariableDeclarationArgument)
+                    .RemoveRange(parametersWithLiteralArgument)
+                    .Where(parameterAndArguments => ParameterNeedsGenerateDeclarationFilter(
+                        inlineMethodRefactoringProvider,
+                        syntaxFacts, semanticModel, parameterAndArguments, cancellationToken))
+                    .ToImmutableArray();
+
+                // 5. Parameter without any argument.
+                // Case 1: only parameter has default value would be left here. The parameter will be replaced by the default value.
+                // Case 2: there is no arguments and the parameter is params array. An array needs to be declared in the caller. Similarly to what is done in step 4
+                // Example for case 2:
+                // Before:
+                // void Caller(bool x)
+                // {
+                //     Callee();
+                // }
+                // void Callee(params int[] a)
+                // {
+                //     DoSomething(a);
+                // }
+                // After:
+                // void Caller(bool x)
+                // {
+                //     int[] a = { };
+                //     DoSomething(a);
+                // }
+                // void Callee(params int[] a)
+                // {
+                //     DoSomething(a);
+                // }
+                var parametersWithoutArgument = calleeMethodSymbol.Parameters.RemoveRange(allParametersWithArgument);
+                if (parametersWithoutArgument.Length == 1 && parametersWithoutArgument[0].IsParams)
+                {
+                    parametersNeedGenerateDeclarations = parametersNeedGenerateDeclarations.Concat((
+                        parametersWithoutArgument[0], ImmutableArray<SyntaxNode>.Empty));
+                }
```

Comment:
why special case the single arg case?  what if there are two args, and the second isparams?

Reply:
Switch to use the IOperation let these checks become unnecessary. :) So mark it as resovled.